spec/busted-log-failed.lua

```diff
@@ -0,0 +1,37 @@
+-- busted-log-failed.lua
+
+-- Log which test files run by busted had failures or errors in a
+-- file.  The file to use for logging is specified in the
+-- FAILED_TEST_FILES_FILE environment variable.  This is used to
+-- reduce test rerun times for flaky tests.
+
+local busted = require 'busted'
+local failed_files_file = assert(os.getenv("FAILED_TEST_FILES_FILE"),
+        "FAILED_TEST_FILES_FILE environment variable not set")
+
+local FAILED_FILES = {}
+
+busted.subscribe({ 'failure' }, function(element, parent, message, debug)
+  if element.trace then
```

Comment:
Should we error out otherwise? Or if it never happens, could we omit this check?